sim/core/dot.go

```diff
@@ -124,12 +125,29 @@ func (dot *Dot) RescheduleNextTick(sim *Simulation) {
 	sim.AddPendingAction(dot.tickAction)
 }
 
+// Snapshots and activates the Dot
+// If the Dot is running it's duration will be refreshed and
+// if there was a next Dot happening this will carry over to the new Dot
 func (dot *Dot) Apply(sim *Simulation) {
+
 	dot.TakeSnapshot(sim, false)
 
-	dot.Cancel(sim)
 	dot.TickCount = 0
-	dot.RecomputeAuraDuration()
+
+	// we a have running dot tick
+	// the next tick never get's clipped and is added onto the dot's time for hasted dots
+	// see: https://github.com/wowsims/cata/issues/50
+	if dot.tickAction != nil {
+
+		// save next tick timer as timer is computed based on tick time
+		// which we update in RecomputeAuraDuration
+		nextTick := dot.TimeUntilNextTick(sim)
+		dot.RecomputeAuraDuration()
+		dot.Aura.Duration += nextTick
```

Comment:
Dont you need to also add an extra tick to the dot? Just increasing its duration will mean it wont tick the last time but just expire I think

Reply:
Yeah I confirmed it's indeed the case. My bad. 

I'd not like to just 'add' an extra tick however. Talents like Improved Devouring Plague require the correct amount of the DoT's ticks. So I see two options.

1. Add a new Option like `HasRollOverTick` which would add +1 in `MaxTicksRemaining`
2. Create a new utility method `ExpectedTickCount` on `Dot` which both `RecomputeAuraDuration` would call and spells / auras that need the dot's tick count. This would have the drawback however, that the tick count could be different if it's queried not on dot apply.

Reply:
I wonder if you can just make the tickCount -1 😅